react/features/calendar-sync/web/microsoftCalendar.js

```diff
@@ -0,0 +1,492 @@
+
+            // not authorized, skip
+            if (!getState()['features/calendar-sync'].msAuthState) {
+                return Promise.resolve();
+            }
+
+            return new Promise((resolve, reject) => {
+                const client = Client.init({
+                    authProvider: done => done(
+                        null,
+                        getState()['features/calendar-sync']
+                            .msAuthState.accessToken)
+                });
+
+                const query = client.api('/me/calendars');
+
+                query.get((error, response) => {
+                    if (error) {
+                        reject(error);
+                    } else {
+                        const calendarIds
+                            = response.value.map(en => en.id);
+                        const promises
+                            = calendarIds.map(id =>
+                                this._getCalendarEntries(
+                                    client, id, fetchStartDays, fetchEndDays));
+
+                        return Promise.all(promises)
+                            .then(result => [].concat(...result))
+                            .then(entries =>
+                                entries.map(
+                                    e => this._convertCalendarEntry(e)))
+                            .then(result => resolve(result));
+                    }
+                });
+            });
+        };
+    }
+
+    /**
+     * Parses the microsoft calendar entries to a known format.
+     *
+     * @param {Object} entry - The microsoft calendar entry.
+     * @returns {{
+     *  id: string,
+     *  startDate: string,
+     *  endDate: string,
+     *  title: string,
+     *  location: string,
+     *  description: string}}
+     * @private
+     */
+    _convertCalendarEntry(entry) {
+        return {
+            id: entry.id,
+            startDate: entry.start.dateTime,
+            endDate: entry.end.dateTime,
+            title: entry.subject,
+            location: entry.location.displayName,
+            description: entry.body.content
+        };
+    }
+
+    /* eslint-disable max-params */
+    /**
+     * Retrieves calendar entries from a specific calendar.
+     *
+     * @param {Object} client - The microsoft-graph-client initialized.
+     * @param {string} calendarId - The calendar ID to use.
+     * @param {number} fetchStartDays - The number of days to go back
+     * when fetching.
+     * @param {number} fetchEndDays - The number of days to fetch.
+     * @returns {Promise<any> | Promise}
+     * @private
+     */
+    _getCalendarEntries(
+            client,
+            calendarId,
+            fetchStartDays: ?number,
+            fetchEndDays: ?number): Promise<*> {
+        /* eslint-enable max-params */
+
+        if (this._isSignedIn()) {
+            this._dispatch(setCalendarAPIState(CALENDAR_API_STATES.SIGNED_IN));
```

Comment:
If you have token that is not expired and you reload the page, this will set the correct state of the calendar-sync that it is signed in, have a correct token ... Maybe this state should be done in one place when we init the api or when we construct the MicrosoftCalendarApi object ..., WDYT?

The fact is that currently when you load the welcome page it calls _fetchCalendarEntries which end up calling getCalendarEntries and I was using that to set the correct state of the feature ...